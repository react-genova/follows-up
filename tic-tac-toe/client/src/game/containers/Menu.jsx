import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Button from '../components/Button';
import { beginGame } from '../modules/engine/action.creators';

const MenuContainer = ({ fireBeginGame }) => (
    <div style={{ flex: 1, display: 'flex', flexDirection: 'column', justifyContent: 'center', alignItems: 'center' }}>
        <Button text="Play again" onClick={fireBeginGame} />
    </div>
);

MenuContainer.propTypes = {
    fireBeginGame: PropTypes.func,
};

MenuContainer.defaultProps = {
    fireBeginGame: undefined,
};

export const mapDispathToProps = {
    fireBeginGame: beginGame,
};

export { MenuContainer };
export default connect(null, mapDispathToProps)(MenuContainer);
