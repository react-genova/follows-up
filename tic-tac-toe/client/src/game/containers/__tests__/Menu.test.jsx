import React from 'react';
import { render, cleanup, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import { MenuContainer } from '../Menu';

describe('Menu conatiner', () => {
    afterEach(() => {
        cleanup();
        jest.clearAllMocks();
    });

    it('renders without crashing', () => {
        const onFireBeginGame = jest.fn();
        const { container, getByText } = render(
            <MenuContainer fireBeginGame={onFireBeginGame} />,
        );
        expect(container.firstChild).toMatchSnapshot();
        fireEvent.click(getByText('Play again'));
        expect(onFireBeginGame).toHaveBeenCalledTimes(1);
    });
});
