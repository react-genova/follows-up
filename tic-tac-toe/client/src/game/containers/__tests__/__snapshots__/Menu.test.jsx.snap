// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu conatiner renders without crashing 1`] = `
<div
  style="display: flex; flex-direction: column; justify-content: center; align-items: center;"
>
  <button
    class="sc-bdVaJa kJcqmF"
    type="button"
  >
    Play again
  </button>
</div>
`;
